onboard/catkin_ws/src/cv/scripts/utils.py

```diff
@@ -53,3 +54,298 @@ def visualize_detections(self, frame, detections):
             self.rectangle(frame_copy, bbox)
 
         return frame_copy
+
+
+class ImageTools(object):
+    def __init__(self):
+        self._cv_bridge = CvBridge()
+
+    def convert_ros_msg_to_cv2(self, ros_data, image_encoding='bgr8'):
+        """
+        Convert from a ROS Image message to a cv2 image.
+        """
+        try:
+            return self._cv_bridge.imgmsg_to_cv2(ros_data, image_encoding)
+        except CvBridgeError as e:
+            if "[16UC1] is not a color format" in str(e) or "[8UC1] is not a color format" in str(e):
+                raise CvBridgeError(
+                    "You may be trying to use a Image method " +
+                    "(Subscriber, Publisher, conversion) on a depth image" +
+                    " message. Original exception: " + str(e))
+            raise e
+
+    def convert_ros_compressed_to_cv2(self, compressed_msg):
+        np_arr = np.fromstring(compressed_msg.data, np.uint8)
+        return cv2.imdecode(np_arr, cv2.IMREAD_COLOR)
+
+    def convert_ros_compressed_msg_to_ros_msg(self, compressed_msg,
+                                              encoding='bgr8'):
+        cv2_img = self.convert_ros_compressed_to_cv2(compressed_msg)
+        ros_img = self._cv_bridge.cv2_to_imgmsg(cv2_img, encoding=encoding)
+        ros_img.header = compressed_msg.header
+        return ros_img
+
+    def convert_cv2_to_ros_msg(self, cv2_data, image_encoding='bgr8'):
+        """
+        Convert from a cv2 image to a ROS Image message.
+        """
+        return self._cv_bridge.cv2_to_imgmsg(cv2_data, image_encoding)
+
+    def convert_cv2_to_ros_compressed_msg(self, cv2_data,
+                                          compressed_format='jpg'):
+        """
+        Convert from cv2 image to ROS CompressedImage.
+        """
+        return self._cv_bridge.cv2_to_compressed_imgmsg(cv2_data,
+                                                        dst_format=compressed_format)
+
+    def convert_ros_msg_to_ros_compressed_msg(self, image,
+                                              image_encoding='bgr8',
+                                              compressed_format="jpg"):
```

Comment:
Inconsistent quotes between ' or ".
I generally don't care, but these are right next to each other